repo/object/object_reader.go

```diff
@@ -102,7 +106,38 @@ func (r *objectReader) Read(buffer []byte) (int, error) {
 	return readBytes, nil
 }
 
+func (r *objectReader) readAhead(ctx context.Context) {
+	if r.opts.ReadAheadBytes <= 0 {
+		return
+	}
+
+	var (
+		contentIDs  []content.ID
+		totalLength int64
+	)
+
+	for i := r.currentChunkIndex + 1; i < len(r.seekTable); i++ {
+		if cid, _, ok := r.seekTable[i].Object.ContentID(); ok {
+			if i > r.maxReadAheadChunkIndex {
+				contentIDs = append(contentIDs, cid)
+				r.maxReadAheadChunkIndex = i
+			}
+
+			totalLength += r.seekTable[i].Length
+			if totalLength > r.opts.ReadAheadBytes {
+				break
+			}
+		}
+	}
+
+	if len(contentIDs) > 0 {
+		go r.cr.PrefetchContents(ctx, contentIDs, "blobs")
+	}
+}
+
 func (r *objectReader) openCurrentChunk() error {
+	r.readAhead(r.ctx)
```

Comment:
Would `openCurrentChunk()` ever be called while there is a previous `PrefetchContents()` still running ? Does this have a worst case where this parallelism could explode on the total number of active fetch threads ?

Reply:
Subsequent reads will generally block on completion of previous blob prefetches and multiple blob prefetches execute only once, so I don't think this will explode - it may create some relatively cheap goroutines though.

The only case I think where it may not behave great is if somebody opens a bunch of large readers, read just one byte and never read the rest, but it's pretty unlikely I'd say?

Having said that we should still have a hard concurrency limit at a lower layer (inside `content.WriteManager`). I'm going to add it.

Reply:
Sounds good.

Also probably add a comment to this path around where the subsequent read will block. This might help readers to understand how that works.